### PR TITLE
[Excalibur] response impact - reduction card data

### LINKED ISSUE #281: [Excalibur] Calculate data to pass into 4 cards for staff/inc hospitalized and fatalities
1. Based on original and current scenarios, determine the reduction in numbers.
2. Hook up to existing UI cards

### PR DESCRIPTION
**Pull request by rocktacular**
## Description of the change

1. Calculate hospitalizations and fatalities across all facilities for current scenario for both staff and incarcerated.
2. repeat for original scenario
3. pass data into cards

Most of the heavy lifting is in the `ResponseImpactCardStateUtils` file. `ResponseImpactDashboard` assembles and maintains the state for the data object. A couple other files I just needed to export some functions, and then the card container I needed to pass those props down and hook the data up.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #281 

### Development

These boxes should be checked by the submitter prior to merging:

- [X] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by rocktacular** [comment]
Here it is hooked up. Looks like there may be a rounding issue sometimes that causes to go to -1.

image

File: src/page-response-impact/utils/ResponseImpactCardStateUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    hospitalized: number;
+    fatalities: number;
+  };
+  staff: {
+    hospitalized: number;
+    fatalities: number;
+  };
+};
+
+export function buildReductionData(
```
Comment (jovergaag) [inline src/page-response-impact/utils/ResponseImpactCardStateUtils.tsx:18]: Beautiful

File: src/page-response-impact/ResponseImpactDashboard.tsx [hunk 2]
```diff
@@ -152,6 +161,22 @@ const ResponseImpactDashboard: React.FC = () => {
     fetchFacilities();
   }, [scenarioState?.data?.id]);
 
+  // calculate data for cards
```
Comment (jovergaag) [inline src/page-response-impact/ResponseImpactDashboard.tsx:164]: This component is starting to get pretty big. I don't think its necessary to do it in this PR but we should probably start thinking about breaking it up.

Comment (rocktacular) [reply src/page-response-impact/ResponseImpactDashboard.tsx:164]: hah for sure. You should have seen it before I moved a couple functions/types to the Utils. 

Depending on the other components' needs, I might be able to move all of this down to the card container component.

**Comment by rocktacular** [comment]
ok to resolve linting issues, I did these 2 things:
1. moved `scenarioState?.data?.id` to its own var `scenarioId`. linter did not like complex (multi-level) objects in the `useEffect` dependency arr.
2. move `fetchFacilities` function into the `useEffect` where it is called. linter asked to the fn into dep array, but then that caused an infinite loop, so linter suggested moving the function def inside the `useEffect`

**Pull request merged**